scripts/gha/build_ios_tvos.py

```diff
@@ -0,0 +1,581 @@
+                         '{0} for os {1}'.format(platform_variant, apple_os))
+
+      archs_from_config = set(os_platform_variant_config['architectures'])
+      supported_archs = archs_from_config.intersection(args.architecture)
+      if not supported_archs:
+        raise ValueError('Could not find valid architectures for platform '
+                         '{0} for os {1}'.format(platform_variant, apple_os))
+
+      for architecture in supported_archs:
+        platform_architecture_token = '{0}-{1}'.format(platform_variant,
+                                                       architecture)
+        archive_output_path = os.path.join(frameworks_os_path,
+                                                platform_architecture_token)
+        # Eg: <build_dir>/tvos_cmake_build/device-arm64
+        build_path = os.path.join(args.build_dir,
+                                  '{0}_cmake_build'.format(apple_os),
+                                  platform_architecture_token)
+        # For ios builds, we specify architecture to cmake configure.
+        architecture = architecture if apple_os == 'ios' else None
+        # For tvos builds, we pass a special cmake option PLATFORM to toolchain.
+        toolchain_platform = os_platform_variant_config['toolchain_platform'] \
+                             if apple_os == 'tvos' else None
+        cmake_configure_and_build(args.source_dir, build_path,
+                                  os_platform_variant_config['toolchain'],
+                                  archive_output_path, supported_targets,
+                                  architecture, toolchain_platform)
+        # Arrange frameworks
+        arrange_frameworks(archive_output_path)
+
+  # if we built for all architectures build universal framework as well.
+  build_universal_framework(frameworks_path)
+
+  # Build xcframeworks
+  xcframeworks_path = os.path.join(args.build_dir, 'xcframeworks')
+  template_info_plist_path = os.path.join(args.source_dir, 'build_scripts',
+                                          'tvos', 'Info_ios_and_tvos.plist')
+  build_xcframeworks(frameworks_path, xcframeworks_path,
+                     template_info_plist_path)
+
+
+def parse_cmdline_args():
+  parser = argparse.ArgumentParser(description='Build for iOS and tvOS.')
+  parser.add_argument('-b', '--build_dir',
+    default='ios_tvos_build', help='Name of build directory.')
+  parser.add_argument('-s', '--source_dir',
+    default=os.getcwd(),
+    help='Directory containing source code (top level CMakeLists.txt)')
+  parser.add_argument('-v', '--platform_variant', nargs='+',
+    default=('device', 'simulator'),
+    help='List of platforms to build for.')
+  parser.add_argument('-a', '--architecture', nargs='+',
+    default=('arm64', 'armv7', 'x86_64', 'i386'),
+    help='List of architectures to build for.')
+  parser.add_argument('-t', '--target', nargs='+',
+    default=('firebase_admob', 'firebase_analytics',
```

Comment:
I wonder if we could reuse the "Config" variable.

Reply:
Good idea but I think having the list here explicitly makes it obvious for users of this standalone script. In addition, the `build_testapps.json` contains config for settings like "test_lab". It might make sense to use that for integration testing.